### PR TITLE
Feature/hf models

### LINKED ISSUE #27: Add HF models support
Create a support to Hugging Face models.
This needs to run a server considering a MODEL_PATH and HF_TOKEN env vars.

### PR DESCRIPTION
**Pull request by antunsz**
#### Changes
- Added a new module named huggingface to support HuggingFace hub.
- Refactored __main__ to enable calling the huggingface module via a shell command.
- Updated the documentation.

Fixes #27 

#### Type of change

- [x] 📚  Documentation Update
- [ ] 🧪 Tests Cases
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [x] 🔬 New feature (non-breaking change which adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 This change requires a documentation update

#### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

File: src/fastserve/models/huggingface.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+class PromptRequest(BaseModel):
+    prompt: str = "Write a python function to resize image to 224x224"
+    temperature: float = 0.8
+    top_p: float = 1.0
+    max_tokens: int = 200
+    stop: List[str] = []
+
+
+class ServeHuggingface(FastServe):
```
Comment (aniketmaurya) [inline src/fastserve/models/huggingface.py:22]: can we rename it to `ServeHuggingFace` please?

File: README.md [hunk 2]
```diff
@@ -102,6 +102,45 @@ app.run_server()
 or, run `python -m fastserve.models --model image-classification --model_name resnet18 --batch_size 4 --timeout 1` from
 terminal.
 
+### Serve HuggingFace Models
+
+You can easily serve any HuggingFace Transformer model using FastServe.
+
+First, ensure you have your HuggingFace API token set up correctly in your environment to access models from HuggingFace Hub.
```
Comment (aniketmaurya) [inline README.md:109]: maybe also mention that this step is only required for some models and not all of them.

**Review by aniketmaurya — COMMENTED** [review]
LGTM! thanks for the great work. I have added some minor comments

File: src/fastserve/models/huggingface.py (hunk 1 above)
Comment (antunsz) [reply src/fastserve/models/huggingface.py:22]: Sure! 👍🏼

File: README.md (hunk 2 above)
Comment (antunsz) [reply README.md:109]: sounds good 👌

Comment (antunsz) [reply README.md]: I did a little change into huggingface.py to not require the token for every model

**Review by aniketmaurya — APPROVED** [review]

**Pull request merged**